test-integration/src/test/java/nl/nn/adapterframework/filesystem/ExchangeFileSystemRuntimeParametersTest.java

```diff
@@ -0,0 +1,55 @@
+package nl.nn.adapterframework.filesystem;
+
+import microsoft.exchange.webservices.data.core.service.item.EmailMessage;
+import microsoft.exchange.webservices.data.property.complex.Attachment;
+import nl.nn.adapterframework.configuration.ConfigurationException;
+import nl.nn.adapterframework.receivers.ExchangeMailListener;
+import nl.nn.adapterframework.testutil.PropertyUtil;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.*;
+
+public class ExchangeFileSystemRuntimeParametersTest  extends MailFileSystemTestBase<EmailMessage, Attachment, ExchangeFileSystem>{
+
+	//private String DEFAULT_URL = "https://outlook.office365.com/EWS/Exchange.asmx";
+
+	private String url         = PropertyUtil.getProperty(PROPERTY_FILE, "url");
+	private String mailaddress = PropertyUtil.getProperty(PROPERTY_FILE, "mailaddress");
+	private String accessToken = PropertyUtil.getProperty(PROPERTY_FILE, "accessToken");
+//	private String basefolder2 = PropertyUtil.getProperty(PROPERTY_FILE, "basefolder2");
+//	private String basefolder3 = PropertyUtil.getProperty(PROPERTY_FILE, "basefolder3");
+
+	private final String SEPARATOR = "|";
+
+
+	private String nonExistingFileName = "AAMkAGNmZTczMWUwLWQ1MDEtNDA3Ny1hNjU4LTlmYTQzNjE0NjJmYgBGAAAAAAALFKqetECyQKQyuRBrRSzgBwDx14SZku4LS5ibCBco+nmXAAAAAAEMAADx14SZku4LS5ibCBco+nmXAABMFuwsAAA=";
+
+	@Override
+	protected ExchangeFileSystem createFileSystem() throws ConfigurationException {
+		ExchangeFileSystem fileSystem = new ExchangeFileSystem();
+		if (StringUtils.isNotEmpty(url)) fileSystem.setUrl(url);
+		fileSystem.setAccessToken(accessToken);
+		fileSystem.setUsername(username);
+		fileSystem.setPassword(password);
+		return fileSystem;
+	}
+
```

Comment:
Zou je ook een tests kunnen toevoegen die laten zien dat je een bericht van een mailbox naar een (folder in een) andere mailbox kan verplaatsen en kopieren?

Reply:
Deze vind ik lastig, om dit te realiseren heb ik een NPA nodig waarmee ik minstens twee mailboxen kan benaderen. Momenteel kan ik enkel met mijn eigen email adres en wachtwoord inloggen om bijbehorende mailbox te openen.

Ik heb gezocht of het mogelijk was dit vorm te geven middels Docker, echter zijn er voor Exchange geen Docker images beschikbaar.

Is het mogelijk om dit op de infrastructuur van WAF te realiseren?